wormhole-connect/src/hooks/useBalanceChecker.ts

```diff
@@ -0,0 +1,73 @@
+import { useEffect, useState, useMemo } from 'react';
+import { amount, routes } from '@wormhole-foundation/sdk';
+import { getWalletConnection, TransferWallet } from 'utils/wallet';
+import config from 'config';
+import { useIsMounted } from './useIsMounted';
+
+export const useBalanceChecker = (
+  quote?: routes.Quote<
+    routes.Options,
+    routes.ValidatedTransferParams<routes.Options>
+  >,
+): {
+  feeSymbol?: string;
+  isCheckingBalance: boolean;
+  hasSufficientBalance: boolean;
+  walletBalance?: number;
+  networkCost?: number;
+} => {
+  const [isCheckingBalance, setCheckingBalance] = useState(false);
+  const [state, setState] = useState<{ balance: number; cost: number } | null>(
+    null,
+  );
+  const isMounted = useIsMounted();
+
+  const feeSymbol = useMemo(() => {
+    if (!quote?.relayFee?.token) return;
+
+    return config.sdkConverter.findTokenConfigV1(
+      quote?.relayFee?.token,
+      Object.values(config.tokens),
+    )?.symbol;
+  }, [quote]);
+
+  useEffect(() => {
+    setCheckingBalance(true);
+    (async () => {
+      try {
+        if (!quote?.relayFee?.amount) return;
+
+        const wallet = getWalletConnection(TransferWallet.SENDING);
+        if (!wallet) return;
+
+        const cost = amount.whole(quote.relayFee.amount);
+
+        const balance = parseFloat(
+          await wallet.getBalance(quote.relayFee.token.address?.toString()),
```

Comment:
What if the relayFee.token is a token that doesn't exist on this chain?

Reply:
We should also handle the case where `relayFee.token` is set to `"native"` (native gas token which has no address)

Reply:
> What if the relayFee.token is a token that doesn't exist on this chain?

there's an early return 8 lines up (line 38) for this case

> We should also handle the case where `relayFee.token.address` is set to `"native"` (native gas token)

added an extra check for this